calyx/src/analysis/read_write_set.rs

```diff
@@ -18,6 +18,60 @@ impl ReadWriteSet {
             .filter(|port| !port.borrow().is_hole())
     }
 
+    /// Returns the "meaningful" [ir::Port] which are read from in the assignments.
+    /// "Meaningful" means we just exclude the following `@done` reads:
+    /// the `@go` signal for the same cell *must* be written to in the group
+    pub fn meaningful_port_read_set<'a>(
```

Comment:
This probably shouldn't be a part of the `ReadWriteSet` analysis itself and be defined in the LRA file?